bundles/mapping/camera-controls-3d/view/CameraControls3d.jsx

```diff
@@ -0,0 +1,141 @@
+const upDownClickHandler = (directionUp) => {
+    const mapmodule = getMapModule();
+    const cam = mapmodule.getCamera();
+    if (directionUp) {
+        cam.location.altitude = cam.location.altitude * ((100 + upDownChangePercent) / 100);
+    } else {
+        cam.location.altitude = cam.location.altitude * ((100 - upDownChangePercent) / 100);
+    }
+    mapmodule.setCamera(cam);
+    Oskari.getSandbox().postRequestByName('MapMoveRequest');
+};
+
+const setCameraToMoveMode = (activeMapMoveMethod) => {
+    if (activeMapMoveMethod === mapMoveMethodMove) {
+        return;
+    }
+    getMapModule().setCameraToMoveMode();
+};
+const setCameraToRotateMode = (activeMapMoveMethod) => {
+    if (activeMapMoveMethod === mapMoveMethodRotate) {
+        return;
+    }
+    getMapModule().setCameraToRotateMode();
+};
+
+const CameraControls3d = ({ mapInMobileMode, getMessage }) => {
+    const [activeMapMoveMethod, setActiveMapMoveMethod] = useState(mapMoveMethodMove);
+
+    const moveMapControl = <MoveMapIcon mapInMobileMode={mapInMobileMode} clickHandler={() => {
+        setActiveMapMoveMethod(mapMoveMethodMove);
+        setCameraToMoveMode();
+    }} title={mapInMobileMode ? '' : getMessage('tooltip.move')} controlIsActive = {activeMapMoveMethod === mapMoveMethodMove}/>;
+
+    const rotateMapControl = <RotateMapIcon mapInMobileMode={mapInMobileMode} clickHandler={() => {
+        setActiveMapMoveMethod(mapMoveMethodRotate);
+        setCameraToRotateMode();
+    }} title={ mapInMobileMode ? '' : getMessage('tooltip.rotate')} controlIsActive = {activeMapMoveMethod === mapMoveMethodRotate}/>;
+
+    const upControl = <UpIcon mapInMobileMode={mapInMobileMode}
+        clickHandler={() => upDownClickHandler(true)} title={mapInMobileMode ? '' : getMessage('tooltip.up')}/>;
+    const downControl = <DownIcon mapInMobileMode={mapInMobileMode}
+        clickHandler={() => upDownClickHandler(false)} title={mapInMobileMode ? '' : getMessage('tooltip.down')}/>;
+
+    let controls;
+
+    if (!mapInMobileMode) {
+        controls = <MapControlsContainer>
+            <MapControlContainer isMobile={mapInMobileMode} controlIsActive = {activeMapMoveMethod === mapMoveMethodMove}>
+                <MapControl>{moveMapControl}</MapControl>
+            </MapControlContainer>
+            <MapControlContainer isMobile={mapInMobileMode} controlIsActive = {activeMapMoveMethod === mapMoveMethodRotate}>
+                <MapControl>{rotateMapControl}</MapControl>
+            </MapControlContainer>
+            <Break/>
+            <MapControlContainer disabled = {activeMapMoveMethod === mapMoveMethodRotate}>
+                <MapControl>{upControl}</MapControl>
+            </MapControlContainer>
+            <MapControlContainer disabled = {activeMapMoveMethod === mapMoveMethodRotate}>
+                <MapControl>{downControl}</MapControl>
+            </MapControlContainer>
+        </MapControlsContainer>;
+    } else {
+        controls = <MobileContainer>
```

Comment:
If you remove the controls variable you can use this as the "default" return statement without the else.

Reply:
Done